### ISSUE TITLE
Ensure ARIA roles, states, and properties are valid

### ISSUE THREAD

**Issue by eelzi-ONRR** [DOI-ONRR/onrr.gov-site#2706]
**OCIO Violation:** Ensure ARIA roles, states, and properties are valid
**Severity**: 5
**Occurrence:** 97%
**Priority:** 7.7

This violation has 17 different notes. All notes have multiple occurrences. See the attached spreadsheet for details on of what the violations are and what pages they are found on.  
[https://app.zenhub.com/files/215585796/444c4434-5232-42a0-b8f6-01c61cea6123/download](https://app.zenhub.com/files/215585796/444c4434-5232-42a0-b8f6-01c61cea6123/download)
Some of them are list items that need to be tagged as menu items instead, which may be help resolve some of the violation in issue #2702.

Notes:

- The role attribute value of 'listitem' given to this A is not allowed.  The element's role attribute should be set to one of the following text values: button | checkbox | menuitem | menuitemcheckbox | menuitemradio | radio | tab | switch | treeitem; or the role attribute can be removed
- This INPUT has a mismatch between the value of its readonly and aria-readonly attributes
- The aria-expanded attribute is not allowed on this DIV
- This SPAN has an id attribute of 'docs-internal-guid-28c214a1-e5e0-1ec8-f292-ee50a89ab429', which is not unique in the same DOM
- This BUTTON has an aria-controls attribute value of 'panel-content-0', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-1', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-2', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-3', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-4', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-5', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-6', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-7', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-8', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-9', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-10', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-11', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-12', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-13', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-14', which includes one or more invalid or duplicate ids
- This BUTTON has an aria-controls attribute value of 'panel-content-15', which includes one or more invalid or duplicate ids

**Comment by eelzi-ONRR** [comment]
Since this now is an problem on 100% of pages it's preventing us from being able to report we have any compliant page at all. We should make this a high priority.

2024 Report
OCIO Violation: Ensure ARIA roles, states, and properties are valid
Severity: 5
Occurrence: 100%
Priority: 10

There are 4 additional causes for this violation that we didn't have last year. I've added these to the list above. I also added the count for each cause.

**Comment by eelzi-ONRR** [comment]
@greensomes I organized the results of the 2024 report for this: https://doimspp.sharepoint.com/:x:/r/sites/NRRD/Shared%20Documents/General/Accessibility/2024_onrr.gov_3330_violations.xlsx?d=w75fae772f40344a28717ec50fcc2cc3a&csf=1&web=1&e=h7mBc2&nav=MTVfezA3NjI0NjZDLUI1NjEtNDY0QS05M0EyLUQ4REUwMzhBQzEzMH0

On the Aria Violation tab the columns represent the aria violation and how many instances of it are on the site, and the rows are the code causing it. You can expand the code rows to show what pages the violation occurs on.

**Comment by greensomes** [comment]
@eelzi-ONRR , I updated the role in the sidemenu template from "listitem" to "menuitem". Most - if not all - of the aria violations should now be resolved in [preview](https://preview-onrr-frontend.app.cloud.gov).

**Comment by greensomes** [comment]
@eelzi-ONRR , a note on the "This BUTTON has an aria-controls attribute value of 'panel-content-0'" violations. These are violations because the Vuetify framework used to construct the expansion panels does not render the expanded content referred to in the button elements until the button is pressed. I looked at the USWDS accordion component and replacing all expansion panels with the USWDS equivalent will resolve the violations. I will create a separate issue to track this.

**Comment by greensomes** [comment]
The following items are remaining:

* This INPUT has a mismatch between the value of its readonly and aria-readonly attributes
* The aria-expanded attribute is not allowed on this DIV
* This SPAN has an id attribute of 'docs-internal-guid-28c214a1-e5e0-1ec8-f292-ee50a89ab429', which is not unique in the same DOM

**Comment by greensomes** [comment]
@eelzi-ONRR, All instances of the following error should be resolved in preview:
> This INPUT has a mismatch between the value of its readonly and aria-readonly attributes

**Comment by greensomes** [comment]
@eelzi-ONRR, I do not see the following issues in the spreadsheet attached to this issue.
> The aria-expanded attribute is not allowed on this DIV

> This SPAN has an id attribute of 'docs-internal-guid-28c214a1-e5e0-1ec8-f292-ee50a89ab429', which is not unique in the same DOM

Can you help me track down where this issues occur? Thanks!

**Comment by greensomes** [comment]
The "aria-expanded" issue will be resolved with the conversion to USWDS accordions.

**Comment by greensomes** [comment]
I have updated `admin/content/content_blocks/1033` with the following JSON that removes the span ids.
```
{
	"time": 1700578117486,
	"blocks": [
		{
			"id": "ReLr5eAxJ6",
			"type": "header",
			"data": {
				"text": "Information Quality",
				"level": 2
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "qsTNZl56xA",
			"type": "paragraph",
			"data": {
				"text": "<span style=\"box-sizing: inherit;\">All information disseminated by ONRR must comply with basic standards of quality to ensure and maximize its objectivity, utility, integrity, and quality. ONRR will ensure that information disseminated will be developed from reliable methods and data sources and will otherwise ensure information quality at each stage of information development. ONRR’s methods for producing quality information will be made transparent, to the maximum extent practicable, through:</span>"
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "m2D20Zp0Oj",
			"type": "list",
			"data": {
				"style": "unordered",
				"items": [
					{
						"content": "<span style=\"box-sizing: inherit;\">Accurate documentation;</span>",
						"items": []
					},
					{
						"content": "<span style=\"box-sizing: inherit;\">Use of appropriate internal and external review procedures;</span>",
						"items": []
					},
					{
						"content": "<span style=\"box-sizing: inherit;\">Consultation with experts and users;</span>",
						"items": []
					},
					{
						"content": "<span style=\"box-sizing: inherit;\">Verification of quality; and</span>",
						"items": []
					},
					{
						"content": "<span style=\"box-sizing: inherit;\">And informing its users on corrections and revisions.</span>",
						"items": []
					}
				]
			}
		},
		{
			"id": "fx4UiWBQiE",
			"type": "paragraph",
			"data": {
				"text": "<span style=\"box-sizing: inherit;\">Information released by ONRR will be developed only from reliable data sources based on accepted practices and policies, utilizing accepted methods for information collection and verification.&nbsp;</span>"
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "vie-ysqtyr",
			"type": "header",
			"data": {
				"text": "Information Quality Act Requirements",
				"level": 3
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "eNo5Vcv1p7",
			"type": "list",
			"data": {
				"style": "unordered",
				"items": [
					{
						"content": "Overall, agencies shall adopt a basic standard of quality (including objectivity, utility, and integrity) as a performance goal and should take appropriate steps to incorporate information quality criteria into agency information dissemination practices. Quality is to be ensured and established at levels appropriate to the nature and timeliness of the information to be disseminated. Agencies shall adopt specific standards of quality that are appropriate for the various categories of information they disseminate.",
						"items": []
					},
					{
						"content": "As a matter of good and effective agency information resources management, agencies shall develop a process for reviewing the quality (including the objectivity, utility, and integrity) of information before it is disseminated. Agencies shall treat information quality as integral to every step of an agency's development of information, including creation, collection, maintenance, and dissemination. This process shall enable the agency to substantiate the quality of the information it has disseminated through documentation or other means appropriate to the information.",
						"items": []
					},
					{
						"content": "To facilitate citizen review, agencies shall establish administrative mechanisms allowing affected persons to seek and obtain, where appropriate, timely correction of information maintained and disseminated by the agency that does not comply with OMB or agency guidelines. These administrative mechanisms shall be flexible, appropriate to the nature and timeliness of the disseminated information and incorporated into agency information resources management and administrative practices.",
						"items": []
					}
				]
			}
		},
		{
			"id": "cgp5AdK7kR",
			"type": "header",
			"data": {
				"text": "Information Quality Directives, Policies, and Guidelines",
				"level": 3
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "QsZi5DZthE",
			"type": "paragraph",
			"data": {
				"text": "<span style=\"box-sizing: inherit;\"></span><span style=\"box-sizing: inherit;\">The following list provides governing authorities for the Information Quality Act.</span>"
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "KltcCJX1oy",
			"type": "list",
			"data": {
				"style": "unordered",
				"items": [
					{
						"content": "<span style=\"box-sizing: inherit;\">Office of Management and Budget (OMB) has issued \"</span><a href=\"https://obamawhitehouse.archives.gov/omb/fedreg_final_information_quality_guidelines/\">Guidelines for Ensuring and Maximizing the Quality, Objectivity, Utility, and Integrity of Information Disseminated by Federal Agencies</a>.\" &nbsp;",
						"items": []
					},
					{
						"content": "<a href=\"https://georgewbush-whitehouse.archives.gov/omb/memoranda/fy2005/m05-03.html\"><span style=\"box-sizing: inherit;\">The Office of Management and Budget directive</span>, Final Information Quality Bulletin for Peer Review, dated December 15, 2004</a>.",
						"items": []
					},
					{
						"content": "<span style=\"box-sizing: inherit;\">Department of the Interior issued<span> </span></span><a aria-label=\"Download PDF\" class=\"onrr-link-pdf\" href=\"https://www.doi.gov/sites/doi.gov/files/uploads/doi_information_quality_guidelines.pdf\" target=\"_blank\">Information Quality Guidelines (PDF)</a><span> </span>Pursuant to Section 515 of the Treasury and General Government Appropriations Act for Fiscal Year 2001.",
						"items": []
					}
				]
			}
		},
		{
			"id": "8QgiYorKuL",
			"type": "header",
			"data": {
				"text": "Additional Resources",
				"level": 3
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "dFKBo2SQhB",
			"type": "list",
			"data": {
				"style": "unordered",
				"items": [
					{
						"content": "<a href=\"/getting-started/how-leasing-works\">How Leasing Works</a>",
						"items": []
					},
					{
						"content": "<a href=\"https://revenuedata.doi.gov/downloads#Data-Collection-and-Validation\">Data Collection and Validation</a>",
						"items": []
					},
					{
						"content": "<a href=\"https://revenuedata.doi.gov/how-revenue-works/audits-and-assurances\">Audits and Assurances</a>",
						"items": []
					},
					{
						"content": "<a href=\"https://www.doi.gov/ocio/information-quality-act-coordinators\">Information Quality Policy Contacts</a>",
						"items": []
					}
				]
			}
		},
		{
			"id": "WgtcqDdsqi",
			"type": "header",
			"data": {
				"text": "Requests for Correction of Information",
				"level": 3
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		},
		{
			"id": "z0DOkb6yxs",
			"type": "paragraph",
			"data": {
				"text": "If you believe that information ONRR has disseminated does not meet the OMB or DOI Information Quality Guidelines, you may email <a href=\"mailto:onrrweb@onrr.gov\">onrrweb@onrr.gov</a> a request for correction."
			},
			"tunes": {
				"alignmentTune": {
					"alignment": "left"
				}
			}
		}
	],
	"version": "2.26.1"
}
```

**Comment by greensomes** [comment]
All environments have been updated with the above JSON content.
